pkg/stash/with_etcd.go

```diff
@@ -0,0 +1,85 @@
+package stash
+
+import (
+	"context"
+	"time"
+
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/observ"
+	"github.com/gomods/athens/pkg/storage"
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/clientv3/concurrency"
+	"golang.org/x/sync/errgroup"
+)
+
+// WithEtcd returns a distributed singleflight
+// using an etcd cluster. If it cannot connect,
+// to any of the endpoints, it will return an error.
+func WithEtcd(endpoints []string, checker storage.Checker) (Wrapper, error) {
+	const op errors.Op = "stash.WithEtcd"
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	c, err := clientv3.New(clientv3.Config{
+		Endpoints:   endpoints,
+		DialTimeout: time.Second * 5,
```

Comment:
why not use c.Timeout?

Reply:
@michalpristas this is just the "init" timeout, so when clientv3 pings the Etcd backend, it only needs 5 seconds to determine whether it's healthy or not, otherwise the c.Timeout could be a minute or two and we don't want to wait that long on Athens Startup